### PR TITLE
Merging 1.4.2 to develop branch

### PR DESCRIPTION
**Pull request by engelsanchez**
@evanmcc @russelldb if you could eyeball the diff. All eunit tests passed for me

### CODE REVIEW

File: src/riak_kv_wm_index.erl [hunk 1]
```diff
@@ -215,11 +215,11 @@ handle_streaming_index_query(RD, Ctx) ->
 
     Opts = riak_index:add_timeout_opt(Timeout, [{max_results, MaxResults}]),
 
-    {ok, ReqID} =  Client:stream_get_index(Bucket, Query, Opts),
-    StreamFun = index_stream_helper(ReqID, Boundary, ReturnTerms, MaxResults, undefined, 0),
+    {ok, ReqID, FSMPid} =  Client:stream_get_index(Bucket, Query, Opts),
+    StreamFun = index_stream_helper(ReqID, FSMPid, Boundary, ReturnTerms, MaxResults, proplists:get_value(timeout, Opts), undefined, 0),
     {{stream, {<<>>, StreamFun}}, CTypeRD, Ctx}.
 
-index_stream_helper(ReqID, Boundary, ReturnTerms, MaxResults, LastResult, Count) ->
```
Comment (russelldb) [inline src/riak_kv_wm_index.erl:222]: This needs moar arguments!!!

Comment (engelsanchez) [inline src/riak_kv_wm_index.erl:222]: Let me get my extra argument generator...

**Comment by russelldb** [comment]
Yeah, looks right. Nice work Mr. Sanchez

**Comment by evanmcc** [comment]
:+1: code looks good, eunit passes other than some known brokenness.

**Pull request merged**